service/controller/v19/resource/cloudformation/update.go

```diff
@@ -50,14 +50,27 @@ func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange inte
 			force := false
 			shutdown := true
 			wait := true
+			instance := ""
+
 			for _, v := range volumes {
 				for _, a := range v.Attachments {
+					instance = a.InstanceID
```

Comment:
I think we have docker volumes also in worker instances so this is just a random instance. Am I missing something?

Reply:
I also think we should ensure to get the instance ID of the master instance we want. Only taking any is not a sound approach, especially when there is such an implementation detail and later something changes and everything breaks and nobody knows why.

Reply:
The search is for instances with docker AND etcd volume. And this is only the master. I didn't touch it and yes I know getting the instanceId is ugly here. All in all I found this code really implicit (master shutdown being 'hidden' in volume detachment) and would like to change this at some point to have proper and explicit handling of the master instance.

Reply:
To be clear I don't think that I am making it worse here.